### ISSUE TITLE
kubectl top takes more than a minute to work

### ISSUE THREAD

**Issue by vishh** [kubernetes/kubernetes#30826]
`kubectl top pod` fails to return any metrics for more than 100s after a pod has been created. 
What is the minimum duration for metrics to be available? 30s (heapster) + [10s (cadvisor] ?
It might be helpful if the output says that metrics will be available soon.
Users might incorrectly think that kubectl top doesn't work.

@kubectl @davidopp

**Comment by AdoHe** [comment]
I think generally top command will take more time to return compared with other commands, we should doc this and put this in top help output.

**Comment by mksalawa** [comment]
AFAIK Heapster's window is 60s. I tried it a few times locally and it needed around 75-80s. (I'm not sure if e.g. complexity of a pod makes a difference here, but I tried a number of simple nginx deployments.)

The message would be helpful, but how about a situation when there are already some "old" pods, with computed metrics, a pod is added and the command is called? Should it show the message as well or not? Or maybe it should show all pods listed and some empty value for those which don't have the metrics yet?
